### ISSUE TITLE
[multicore-sys-monitor@ccadeptic23] Stopped working after upgrade to Mint 19.3

### ISSUE THREAD

**Issue by sphh** [linuxmint/cinnamon-spices-applets#2768]
```
 * Applet version/Build date: Version 1.8.91 (2019-11-25 11:03:58)
 * Cinnamon version: Cinnamon 4.4.5
 * Distribution: Mint 19.3
 * Graphics hardware *and* driver used
 * Snippet or attachment of ~/.cinnamon/glass.log
 * Snippet or attachment of ~/.xsession-errors

 ```

**Notify author of applet**
@ccadeptic23
@jaszhix 
@brownsr 

**Issue**
There is a rather good description in the comments section: https://cinnamon-spices.linuxmint.com/applets/view/79#comments:

v1.8.91 Appears to have trouble on Mint 19.3 with Cinnamon 4.4.5. CPU stops updating after a few minutes and I don't think I've noticed MEM update after initial start either. Network and Disk both stop updating and preferences is inaccessible shortly after CPU stops updating. Applet never crashes completely and does launch the system monitor correctly when clicked. I did upgrade from 19.2, and followed the steps to remove configs and cached files.

**Other information**

.xsession-errors
```
anonymous/formatBytes@/home/sph/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/4.0/DataProviders.js:35:44
anonymous/NetDataProvider.prototype.getTooltipString@/home/sph/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/4.0/DataProviders.js:286:18
anonymous/MyApplet.prototype._update@/home/sph/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/4.0/applet.js:331:30
anonymous/MyApplet.prototype.__init/this.loopId<@/home/sph/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/4.0/applet.js:243:86
```

**Comment by claudiux** [comment]
@sphh What is your kernel version? `uname -mr`

**Comment by sphh** [comment]
`$ uname -mr`
`5.1.15-surface-linux-surface x86_64`

**Comment by trourance** [comment]
Somehow the same issue with arch linux and cinnamon 4.4.4. The applet is frozen and I get the same error message in .xsession-errors:
```
(cinnamon:1095): Cjs-WARNING **: 09:39:38.417: JS ERROR: TypeError: sizes[i] is undefined
anonymous/formatBytes@/home/fgos/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/4.0/DataProviders.js:35:44
anonymous/NetDataProvider.prototype.getTooltipString@/home/fgos/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/4.0/DataProviders.js:286:18
anonymous/MyApplet.prototype._update@/home/fgos/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/4.0/applet.js:331:30
anonymous/MyApplet.prototype.__init/this.loopId<@/home/fgos/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/4.0/applet.js:243:86
```
```
$ uname -r
5.4.2-arch1-1
```

**Comment by claudiux** [comment]
```
$ uname -mr
4.15.0-72-generic x86_64
```
Using Mint 19.3 and Cinnamon 4.4.5 with a kernel lesser than 5, this applet works well. I'll try with a kernel version 5.

**Comment by claudiux** [comment]
```
$ uname -mr
5.3.0-24-generic x86_64
```
This applet seems to be working well. Can you specify "a few minutes"? Is the computer idle? Has the screen saver been activated?

**Comment by trourance** [comment]
Today it has frozen few seconds after the login. The computer was in use and the screen saver has never been activated during that time.

**Comment by claudiux** [comment]
Do you also use one of these applets?

turbonote@iksws.com.b 

vnstat@linuxmint.com

netusagemonitor@pdcurtis

**Comment by trourance** [comment]
None of them. On my side, I use ScreenShot@tech71 and temperature@fevimu.

**Comment by claudiux** [comment]
@trourance @sphh Please try replacing the line 31 of `DataProviders.js` by this one:

```
let i = Math.max(0, Math.floor(Math.log(bytes) / Math.log(k)));
```

**Comment by claudiux** [comment]
Sorry, a value calculation error is now present. It will be corrected in #2781.

**Comment by sphh** [comment]
Thanks @claudiux for investigating and finding this!!!
